src/classes/CW_BatchAudienceGroupRelationAssignTest.cls

```diff
@@ -0,0 +1,62 @@
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AGTest'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test_T'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test______T'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test______C'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test____T______C'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test____T______T'));            
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'AG_Test'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'_T'));
+            audienceGroupRecords.AddAll(TestDataUtil.createAudienceGroups(campaignRecord.Id,1,false,'_C'));
+
+            campaignIds.add(campaignRecord.Id);
+        }
+        for(CW_Audience_Group__c audienceGroupRecord:audienceGroupRecords)
+        {
+            audienceGroupRecord.Name = audienceGroupRecord.Name.substring(0,audienceGroupRecord.Name.length()-1);
+        }
+        insert audienceGroupRecords;
+
+        CW_BatchAudienceGroupRelationAssign batchJob1 = new CW_BatchAudienceGroupRelationAssign();
+        CW_BatchAudienceGroupRelationAssign batchJob2 = new CW_BatchAudienceGroupRelationAssign(ConstantUtil.AUDIENCE_CONTROL_GROUP_SUFFIX_VALUE, ConstantUtil.AUDIENCE_TARGET_GROUP_SUFFIX_VALUE);
+        CW_BatchAudienceGroupRelationAssign batchJob3 = new CW_BatchAudienceGroupRelationAssign(ConstantUtil.AUDIENCE_CONTROL_GROUP_SUFFIX_VALUE, ConstantUtil.AUDIENCE_TARGET_GROUP_SUFFIX_VALUE, campaignIds);
+
+        Test.StartTest();
+        Id batchJobId1 = Database.executeBatch(batchJob1);
+        Id batchJobId2 = Database.executeBatch(batchJob2);
+        Id batchJobId3 = Database.executeBatch(batchJob3);
+        Test.StopTest();
+        System.AssertEquals(true, [SELECT Id, Is_Control_Group__c from CW_Audience_Group__c WHERE Name='AGTest_C'][0].Is_Control_Group__c);
+        System.AssertEquals(true, [SELECT Id, Is_Control_Group__c from CW_Audience_Group__c WHERE Name='AG_Test______C'][0].Is_Control_Group__c); 
+        System.AssertEquals(true, [SELECT Id, Is_Control_Group__c from CW_Audience_Group__c WHERE Name='AG_Test____T______C'][0].Is_Control_Group__c); 
+        
+        System.AssertEquals([SELECT Id from CW_Audience_Group__c WHERE Name='AGTest_C'][0].Id, [SELECT Id, Control_Group__c from CW_Audience_Group__c WHERE Name='AGTest_T'][0].Control_Group__c);
+        System.AssertEquals([SELECT Id from CW_Audience_Group__c WHERE Name='AG_Test______C'][0].Id, [SELECT Id, Control_Group__c from CW_Audience_Group__c WHERE Name='AG_Test______T'][0].Control_Group__c); 
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.. Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.